sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/BeamFnDataInboundObserver2.java

```diff
@@ -179,6 +118,69 @@ public void awaitCompletion() throws Exception {
     }
   }
 
+  /** Dispatches the data and timers from the elements to corresponding receivers. */
+  public void multiplexElements(Elements elements) throws Exception {
```

Comment:
can this return true/false if we are done and in ProcessBundleHandler throw an error if we aren't done.

Reply:
it is done if no exception is thrown from the function, and if it is not done the exception should be propagated to ProcessBundleHandler.

Reply:
I think this depends on whether we still want to send the terminal elements from runner if we know from the runner that the bundle is completed with the embedded elements, in which case the endpoint status tracking seems not useful.

Reply:
The purpose was to ensure that a malformed message from the runner lead to an error instead of the bundle silently completing.